src/Cli/ExtraCommands.php

```diff
@@ -49,6 +50,138 @@ public function repoConvertData($path, $options = ['format' => 'json'])
         return $reposResult;
     }
 
+    /**
+     * @command project:update-info
+     * @param $project The github project to update info for.
+     *
+     * @return Consolidation\OutputFormatters\StructuredData\RowsOfFields
+     */
+    public function projectUpdateInfo($project, $options = [
+        'as' => 'default',
+        'codeowners' => '',
+        'support-level-badge' => '',
+        'branch-name' => 'project-update-info',
+        'commit-message' => 'Update project information.',
+        'pr-title' => 'Update project information.',
+        'pr-body' => '',
+        'base-branch' => 'master',
+    ])
+    {
+        if (count(explode('/', $project)) != 2) {
+            throw new \Exception("Invalid project name: $project");
+        }
+        if (empty($options['codeowners']) && empty($options['support-level-badge'])) {
+            throw new \Exception("Must specify at least one of --codeowners or --support-level-badge");
+        }
+        $url = "git@github.com:$project.git";
+        $remote = new Remote($url);
+        $api = $this->api($options['as']);
+        $dir = sys_get_temp_dir() . '/hubph/' . $remote->project();
+        $baseBranch = $options['base-branch'];
+        $workingCopy = WorkingCopy::cloneBranch($url, $dir, $baseBranch, $api);
+
+        // Set logger for workingCopy, is this correct?
+        if ($this->logger) {
+            $workingCopy->setLogger($this->logger);
+        }
+
+        $branchName = $options['branch-name'];
+        $workingCopy->createBranch($branchName);
+        $workingCopy->switchBranch($branchName);
+
+        if (!empty($options['codeowners'])) {
+            $codeowners = $options['codeowners'];
+            // Append given CODEOWNERS line.
+            file_put_contents("$dir/CODEOWNERS", '* ' . $codeowners . "\n", FILE_APPEND);
+            $workingCopy->add("$dir/CODEOWNERS");
+        }
+
+        if (!empty($options['support-level-badge'])) {
+            $support_level_badge = $options['support-level-badge'];
+            $badge_contents = $this->getSupportLevelBadge($support_level_badge);
+            if (!$badge_contents) {
+                throw new \Exception("Invalid support level badge: $support_level_badge");
+            }
+            if (file_exists("$dir/README.md")) {
+                $readme_contents = file_get_contents("$dir/README.md");
+            }
+            else {
+                $readme_contents = '';
+            }
+            $lines = explode("\n", $readme_contents);
```

Comment:
It might be easier to use multiline preg_replace / preg_match for some of this, e.g. putting the support level badge before the license / detecting if there are no licenses.  It might be complicated to make a regex to put the support level badge after the last badge though. You'd need an expression that matched all of the badges.

Maybe it's easier to just keep the loop. I'd break it out into a separate function, though.

Reply:
Moved to a separate function.